buildkite/resource_organization.go

```diff
@@ -0,0 +1,267 @@
+package buildkite
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	resource_schema "github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type organizationResourceModel struct {
+	AllowedApiIpAddresses types.List   `tfsdk:"allowed_api_ip_addresses"`
+	ID                    types.String `tfsdk:"id"`
+	UUID                  types.String `tfsdk:"uuid"`
+}
+
+type organizationResource struct {
+	client *Client
+}
+
+func newOrganizationResource() resource.Resource {
+	return &organizationResource{}
+}
+
+func (organizationResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_organization"
+}
+
+func (o *organizationResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	o.client = req.ProviderData.(*Client)
+}
+
+func (*organizationResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = resource_schema.Schema{
+		Attributes: map[string]resource_schema.Attribute{
+			"id": resource_schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"uuid": resource_schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"allowed_api_ip_addresses": resource_schema.ListAttribute{
+				Optional:    true,
+				ElementType: types.StringType,
+			},
+		},
+	}
+}
+
+func (o *organizationResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan, state organizationResourceModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Get Organization
+	response, err := getOrganization(o.client.genqlient, o.client.organization)
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to obtain Organization",
+			fmt.Sprintf("Unable to obtain Organization: %s", err.Error()),
+		)
+		return
+	}
+
+	if response.Organization.Id == "" {
+		resp.Diagnostics.AddError(
+			"Organization not found",
+			fmt.Sprintf("Organization not found: %s", err.Error()),
+		)
+		return
+	}
```

Comment:
These errors will surface from the request on line 98 itself so to save an extra request slowing down the provider we can remove these. There's a few below in the other functions as well